File: app/static/overlay/js/components/splash-screen.js
Quartz.registerComponent('splash-screen', {
	QZDOM: function(){
		return `<splash-screen class="translucid">
			<div class="bg"></div>

			<div class="music-wrapper">
				<div class="equalizer-icon"><div></div><div></div><div></div><div></div></div>
				<span class="song">WikiLeaks and the Need for Free Speech</span> <span class="artist">&ndash; Dan Bull</span>
			</div>

			<div class="splash-name">
				<span class="thin">parti</span><span class="bold">pirate</span><span class="thin">.org</span>
			</div>

			<div class="splash-text">
				On arrive très vite.<br>
				00:00:00
			</div>


		</splash-screen>`;
	},
	_active: false,
	_settings: {
		'translucid': false,
		'music': true,
		'text': 'On arrive très vite.',
		'countdown': 0, //todo date
		'name': '<span class="thin">parti</span><span class="bold">pirate</span><span class="thin">.org</span>'
	},
	QZinit: function(){
		this.QZ.info('initialized');

		this.toggleSplash(true);
		this.QZ.interval('flip', 4*1000, function(){
			this.toggleSplash(!this._active);
		});

		this.QZ.on('settings', function(data){
			this._settings = data;
			this.updateDisplay();
		});

		this.QZ.send('get-settings');
		this.updateDisplay();
	},
	updateDisplay: function(){
		var domElm = document.querySelector('splash-screen');
		if (this._settings.translucid)
			domElm.classList.add('translucid');
		else
			domElm.classList.remove('translucid');

		if (this._settings.music){
			domElm.querySelector('.music-wrapper').classList.add('visible');
		}
		else {
			domElm.querySelector('.music-wrapper').classList.remove('visible');
		}
		// TODO start or stop music if splash is active and depending on settings and current playing state

		domElm.querySelector('.splash-text').innerHTML = `${this._settings.text}<br>00:00:00`;
		domElm.querySelector('.splash-name').innerHTML = this._settings.name;
	},
	toggleSplash: function(active){
		this._active = active;

		if (active){
			document.querySelector('splash-screen').classList.add('active');
		} else {
			document.querySelector('splash-screen').classList.remove('active');
		}

		this.QZ.broadcastEvent('splash', {'active': active});
	},
});



/*


*/

File: app/modules/components/SplashScreen.py
from . import _QZComponent

class SplashScreen(_QZComponent.QZComponent):
	"""docstring for SplashScreen"""
	def __init__(self):
		super(SplashScreen, self).__init__(QZCOMPONENT['name'])

		# Initial state
		self.QZset_state({
			"active": False,
			"translucid": False,
			"music": False,
			"text": 'On arrive très vite.',
			"countdown": "",
			"name": '<span class="thin">parti</span><span class="bold">pirate</span><span class="thin">.org</span>'
		})

		# Handlers
		def fn(data):
			# A bit hackish but we don't want to push the state change to every client for the toggle, because they get a toggle event already.
			self.QZget_state()['active'] = data['active']

			self.QZsend('toggle', {'active': data['active']})
		self.QZon('toggle_request', fn)
		

QZCOMPONENT = {
	"name": "splash-screen",
	"class": SplashScreen
}
